lfs/transfer_queue.go

```diff
@@ -231,7 +276,12 @@ func (q *TransferQueue) transferWorker() {
 		}
 
 		if err := transfer.Transfer(cb); err != nil {
-			q.errorc <- err
+			if q.canRetry(err) {
```

Comment:
Perhaps we could extract something to wrap this sort of behavior? At a glance, I see it in three places in this file. Perhaps such a thing may be overkill, but I'm curious of your thoughts.